packages/ra-core/src/dataProvider/useDelete.ts

```diff
@@ -330,13 +318,6 @@ export const useDelete = <
             mode.current = mutationMode;
         }
 
-        if (mode.current === 'pessimistic') {
```

Comment:
moved down to allow snapshot (defined L350) to be accessible in pessimistic mode.